src/operator/controllers/intents_reconcilers/kafka_acls_test.go

```diff
@@ -0,0 +1,328 @@
+	"github.com/otterize/intents-operator/src/shared/testbase"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"path/filepath"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"testing"
+)
+
+const (
+	kafkaServiceName  string = "kafka"
+	kafkaTopicName    string = "test-topic"
+	clientName        string = "test-client"
+	intentsObjectName string = "test-client-intents"
+	usernameMapping   string = "user-name-mapping-test"
+)
+
+type KafkaACLReconcilerTestSuite struct {
+	testbase.ControllerManagerTestSuiteBase
+	Reconciler     *KafkaACLReconciler
+	mockKafkaAdmin *kafkaaclsmocks.MockClusterAdmin
+	clientName     string
+}
+
+func (s *KafkaACLReconcilerTestSuite) SetupSuite() {
+	s.TestEnv = &envtest.Environment{}
+	var err error
+	s.TestEnv.CRDDirectoryPaths = []string{filepath.Join("..", "..", "config", "crd")}
+
+	s.RestConfig, err = s.TestEnv.Start()
+	s.Require().NoError(err)
+	s.Require().NotNil(s.RestConfig)
+
+	s.K8sDirectClient, err = kubernetes.NewForConfig(s.RestConfig)
+	s.Require().NoError(err)
+	s.Require().NotNil(s.K8sDirectClient)
+
+	err = otterizev1alpha1.AddToScheme(s.TestEnv.Scheme)
+	s.Require().NoError(err)
+}
+
+func (s *KafkaACLReconcilerTestSuite) SetupTest() {
+	s.ControllerManagerTestSuiteBase.SetupTest()
+}
+
+func (s *KafkaACLReconcilerTestSuite) setupServerStore(serviceName string) *kafkaacls.ServersStore {
+	serverConfig := &otterizev1alpha1.KafkaServerConfig{
+		Spec: otterizev1alpha1.KafkaServerConfigSpec{
+			Service: otterizev1alpha1.Service{
+				Name: serviceName,
+			},
+			Topics: []otterizev1alpha1.TopicConfig{{
+				Topic:                  "*",
+				Pattern:                otterizev1alpha1.ResourcePatternTypePrefix,
+				ClientIdentityRequired: false,
+				IntentsRequired:        false,
+			},
+			},
+		},
+	}
+
+	serverConfig.SetNamespace(s.TestNamespace)
+	emptyTls := otterizev1alpha1.TLSSource{}
+	kafkaServersStore := kafkaacls.NewServersStore(emptyTls, true)
+	kafkaServersStore.Add(serverConfig)
+	return kafkaServersStore
+}
+
+func (s *KafkaACLReconcilerTestSuite) TearDownSuite() {
+	GetNewKafkaIntentsAdmin = kafkaacls.NewKafkaIntentsAdmin
+	s.ControllerManagerTestSuiteBase.TearDownSuite()
+}
+
+func (s *KafkaACLReconcilerTestSuite) BeforeTest(_, testName string) {
+	s.ControllerManagerTestSuiteBase.BeforeTest("", testName)
+
+	controller := gomock.NewController(s.T())
+	s.mockKafkaAdmin = kafkaaclsmocks.NewMockClusterAdmin(controller)
+
+	s.initKafkaIntentsAdmin(true)
+}
+
+func (s *KafkaACLReconcilerTestSuite) initKafkaIntentsAdmin(enableAclCreation bool) {
+	kafkaServersStore := s.setupServerStore(kafkaServiceName)
+	GetNewKafkaIntentsAdmin = func(serverConfig otterizev1alpha1.KafkaServerConfig, _ otterizev1alpha1.TLSSource, enableKafkaACLCreation bool) (kafkaacls.KafkaIntentsAdmin, error) {
```

Comment:
Instead of having this global variable, pass the factory function to `NewKafkaACLReconciler`.